src/axl_async_bbapi.c

```diff
@@ -371,12 +396,32 @@ int __axl_async_start_bbapi (int id, int resume) {
     }
 #endif
 
-    /* Launch the transfer */
+    /*
+     * Launch the transfer.  Note, while BB_StartTransfer() does launch an
+     * asynchronous transfer, that doesn't mean it returns immediately.  For
+     * example, when launched as a job, it can take the following amount of time
+     * for the function to return:
+     *
+     * Amount   Startup time
+     * ------   ------------
+     * 1GB      1 sec
+     * 10GB     5 sec
+     * 20GB     10 sec
+     * 100GB+   17 sec
+     *
+     * And for whatever reason, BB_StartTransfer() returns almost immediately
+     * on interactive nodes.  A 20GB BB_StartTransfer() returns instantly, while
+     * 100GB+ returns in 1-2 sec.
+     */
     int rc = BB_StartTransfer(tdef, thandle);
+
+    /* mark this transfer as in progress */
+    kvtree_util_set_int(file_list, AXL_KEY_STATUS, AXL_STATUS_INPROG);
```

Comment:
Let's move this ``INPROG`` step to come after the ``bb_check(rc)`` block.

Reply:
Fixed in latest push